### PR TITLE
fix: 5.x Update NSG handling, optionally force creation

### PR DESCRIPTION
**Pull request by devoncrouse**
- Support `create_nsgs_always` to force creation e.g. when applying only network resources and bastion.
- Minor fix for worker `nsg_ids` default/pool configuration.
- fix: No utilities for disabled cluster, operator
- fix: Update DRG attachment specification
- fix: Update NSG handling, optionally force creation
- fix: Update operator, extension lifecycle change detection
- style: Update .gitignore
- fix: Create NSG rule for operator to CP if create_nsgs_always
- fix: Dynamic NSG rules
- fix: Output ssh_to_bastion with provided address
- fix: NPN NSG rule conditional, add extra preconditions

### CODE REVIEW

File: module-network.tf [hunk 1]
```diff
@@ -83,6 +83,7 @@ module "network" {
   create_bastion               = var.create_bastion
   create_fss                   = var.create_fss
   create_nsgs                  = var.create_nsgs
```
Comment (hyder) [inline module-network.tf:85]: what's the difference between `create_nsgs` and `create_nsgs_always`?

Comment (devoncrouse) [reply module-network.tf:85]: This is similar to the subnets `never`/`auto`/`always` use-case, and I added a comment that we might consider aligning on that for NSGs.

NSGs are created by default when called for, e.g. operator NSG when operator is enabled, Pods with NPN cluster, etc. In cases where network resources are created separately and re-used, `create_nsgs_always` forces creation even when the related components are disabled.

File: modules/network/nsg-pods.tf
Comment (hyder) [inline modules/network/nsg-pods.tf:1]:
``` "Allow ALL egress from pods to internet" : {
      protocol = local.all_protocols, port = local.all_ports, destination = local.anywhere, destination_type = local.rule_type_cidr,
      enabled  = var.allow_pod_internet_access,
    },```

This should be explicitly controlled IMO.

File: modules/network/nsg-workers.tf
Comment (hyder) [inline modules/network/nsg-workers.tf:1]: SSH to workers should go through bastion even though the workers are public

Comment (hyder) [reply modules/network/nsg-workers.tf:1]: [Line 53-](https://github.com/devoncrouse/terraform-oci-oke/blob/fix-nsgs/modules/network/nsg-workers.tf#L53)[Line 64](https://github.com/devoncrouse/terraform-oci-oke/blob/fix-nsgs/modules/network/nsg-workers.tf#L64)

**Review by hyder — CHANGES_REQUESTED** [review]
changes related to ssh

**Review by hyder — APPROVED** [review]
approved

**Pull request merged**